### DISCUSSION TITLE
Issues Creating Task Triggers

### DISCUSSION THREAD

**Discussion by nockscitney** [dahall/TaskScheduler#942] [Q&A] [answered]
Hi,

Would anyone be able to point me in the right directions with this.  If I try to create anything other than a TimeTrigger (i.e. DailyTrigger), when the task is registered I get a warning in my Task Scheduler logging as follows:

"_Task registered task "\MyTask" but not all specified triggers will start the task. User Action: Ensure that all the task triggers are valid as configured. Additional Data: Error Value: 2147942487._"

I've been going through everything with a fine tooth comb, but I can't find the reason.  I initially thought that it was because I was creating my trigger in a separate method, however splitting it out, in-line yields the same results.

For reference, here is my trigger function:

```
private DailyTrigger CreateDailyTrigger(TaskSchedulerModel taskScheduler)
   {
      DailyTrigger dt = new();

      dt.Id = $"{ taskScheduler.TaskName } Daily Trigger";
      dt.Enabled = true;
      dt.StartBoundary = DateTime.Parse($"{ DateOnly.FromDateTime(DateTime.Now) } { TimeOnly.FromDateTime(taskScheduler.TaskStartTime) }");
      dt.EndBoundary = DateTime.MaxValue;
      dt.Repetition = new RepetitionPattern(taskScheduler.IntervalTimeSpan(), taskScheduler.DurationTimeSpan(), false);
      dt.DaysInterval = (short)taskScheduler.IntervalLength;

      return dt;
   }
```

and it's been set to the task definition using the following switch statement (which again, I removed and didn't check for the trigger type and still ended up with the same result):

```
object tt = taskScheduler.TaskType switch
         {
            TaskType.Daily => td.Triggers.Add(CreateDailyTrigger(taskScheduler)),
            TaskType.Weekly => td.Triggers.Add(CreateWeeklyTrigger(taskScheduler)),
            TaskType.Monthly => td.Triggers.Add(CreateMontlyTrigger(taskScheduler)),
            _ => td.Triggers.Add(CreateDailyTrigger(taskScheduler))
         };
```

Any pointers would be gratefully appreciated.

**Comment by nockscitney** [comment]
UPDATE:

I've been working through the issue today and it seems to be not related to the trigger at all, instead it is happening when I try to set the user that the task is going to run as.

The first call works and creates the task successfully:
`ts.RootFolder.RegisterTaskDefinition(taskScheduler.TaskName, td);`

However, this is what I was using and is causing the failure:
`ts.RootFolder.RegisterTaskDefinition(taskScheduler.TaskName, td, TaskCreation.CreateOrUpdate, userID, password, TaskLogonType.InteractiveTokenOrPassword);`

Currently investigating other ways to do this to see if I can get round it or not

**Comment by dahall** [comment, marked as answer]
You cannot supply a username or password when using `TaskLogonType.InteractiveTokenOrPassword` if the task was first registered using `TaskLogonType.InteractiveToken` which is the default when calling `ts.RootFolder.RegisterTaskDefinition(taskScheduler.TaskName, td);`. -- Microsoft limitation. I'd try using the full parameter overload for `RegisterTaskDefinition` the first time like you have tried to do subsequently.
